### ISSUE TITLE
v3.0.0-beta4: in macro annotation, `new Foo[Int]{}` converted into `new Foo[Int]()`

### ISSUE THREAD

**Issue by mandubian** [scalameta/paradise#144]
```scala
trait Foo[A]
@mymacro val foo = new Foo[Int]{}

// in macro, using q"$valn"
// is converted into val foo = new Foo[Int]()
```

**Comment by olafurpg** [comment]
Thanks for reporting! I can confirm this is a bug in the converter, it appears that an `Apply` node is lost in templates:

```scala
Apply(AppliedTypeTree(Ident(TypeName("Foo")), List(Ident(TypeName("T")))), List()) // before entering template
AppliedTypeTree(Ident(TypeName("Foo")), List(Ident(TypeName("T")))) // after template

```

I suspect #141 is related. I'll need more time to investigate this.

**Comment by olafurpg** [comment]
You can try out the fix with version `3.0.0.145` via the scala.meta bintray ivy repo once the ci completes after the merge on master. Please let me know how it works!
